ledger/acctupdates.go

```diff
@@ -1497,6 +1502,82 @@ func (au *accountUpdates) lookupImpl(rnd basics.Round, addr basics.Address, with
 			}
 		}
 
+		au.accountsMu.RUnlock()
+		needUnlock = false
+
+		var dbRound basics.Round
+		// No updates of this account in the in-memory deltas; use on-disk DB.
+		// The check in roundOffset() made sure the round is exactly the one
+		// present in the on-disk DB.  As an optimization, we avoid creating
+		// a separate transaction here, and directly use a prepared SQL query
+		// against the database.
+		data, dbRound, err = au.accountsq.lookup(addr)
+		if dbRound == currentDbRound {
+			return data, err
+		}
```

Comment:
this code is run without a lock, what would happen if another thread adds an update to in-memory? Do we return outdated info from the DB?

Reply:
yes, another thread could make a concurrent memory update. however, the query is always for historical data, so as long as we can guarantee that the account data is the correct one for the given round, it doesn't really matter.

Reply:
I.e. the case where we return an error since the round is not in the [dbround..dbround+len(delta)], is not an issue since it applied regardless of whether we have a lock or not.